controller/argument_value_resolver.rst

```diff
@@ -51,12 +55,12 @@ controller::
 
     namespace AppBundle\Controller;
 
-    use AppBundle\Entity\User;
     use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\Security\Core\User\UserInterface;
 
     class UserController
     {
-        public function indexAction(User $user)
+        public function indexAction(UserInterface $user)
```

Comment:
That is correct however the current example will not work. 
It throws the following exception "Unable to guess how to get a Doctrine instance from the request information for parameter "user"."
It's because as you noted the param converters execute first. The DoctrineParamConverter=>supports() returns true, because it is a doctrine entity. Then the DoctrineParamConverter=>apply() method then throws the exception.
The security bundle SecurityUserValueResolver is never executed.